### PR TITLE
Add Windows build

### PR DESCRIPTION
**Pull request by mmcauliffe**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mmcauliffe** [comment]
The patch has some changes that could be merged into upstream to make things easier.  The enum  for CDRewriteMode has `OPTIONAL` which was throwing a compiler error, so I've renamed them to `MODE_OPTIONAL` and `MODE_OBLIGATORY`, and made them scoped enums  (I don't know C++ well enough to know if that's the right way to have them, but I was following the style in OpenFst).

File: recipe/patches/0001-Add-Windows-build.patch [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
++++ extensions/cdrewrite.h
+@@ -49,9 +49,9 @@
+ 
+ namespace fst {
+ 
+-enum CDRewriteDirection { LEFT_TO_RIGHT, RIGHT_TO_LEFT, SIMULTANEOUS };
++enum class CDRewriteDirection { LEFT_TO_RIGHT, RIGHT_TO_LEFT, SIMULTANEOUS };
+ 
+-enum CDRewriteMode { OBLIGATORY, OPTIONAL };
++enum class CDRewriteMode { MODE_OBLIGATORY, MODE_OPTIONAL };
```
Comment (kylebgorman) [inline recipe/patches/0001-Add-Windows-build.patch:25]: I'm happy to do a `post` release where these become enum classes. What's the point of `MODE_` prefix, is there something else in the namespace conflicting?

**Review by kylebgorman — COMMENTED** [review]
Few minor notes here.

File: recipe/patches/0001-Add-Windows-build.patch (hunk 1 above)
Comment (mmcauliffe) [reply recipe/patches/0001-Add-Windows-build.patch:25]: Yeah, plain "OPTIONAL" throws a syntax error:

```
C:\Users\michael\miniconda3\conda-bld\pynini_1658677198207\work\extensions\cdrewrite.h(419): error C2760: syntax error: unexpected token ':', expected 'expression'
  C:\Users\michael\miniconda3\conda-bld\pynini_1658677198207\work\extensions\cdrewrite.h(581): error C2760: syntax error: unexpected token ':', expected 'expression'
```

Comment (kylebgorman) [reply recipe/patches/0001-Add-Windows-build.patch:25]: Wow, trashy.

Comment (mmcauliffe) [reply recipe/patches/0001-Add-Windows-build.patch:25]: Yeah, the only reference I can find is to "std::optional", but I don't think that would trigger this? Building on Windows is always an adventure...

**Pull request merged**